packages/shared/src/stateChainGateway/index.ts

```diff
@@ -51,23 +51,25 @@ export const fundStateChainAccount = async (
     amount,
     extractOverrides(txOpts),
   );
+  await transaction.wait(txOpts.wait);
 
-  return transaction.wait(txOpts.wait) as Promise<ContractTransactionReceipt>;
```

Comment:
`transaction.wait()` will return null if `txOpts.wait` is 0, because transaction receipts are only available after the tx was included

Reply:
instead of returning the receipt, we can return the transaction response which will be available even before the tx is included in a block

Reply:
the tx response will already include the tx hash (which is the only thing we use in the sdk). we can change these methods because they are not exposed from the sdk directly